packages/website/src/docs/extending-cadl/emitters-basics.md

```diff
@@ -0,0 +1,155 @@
+---
+eleventyNavigation:
+  key: Writing Emitters
+  parent: Extending Cadl
+  order: 2
+---
+
+# Writing emitters
+
+Cadl emitters are libraries that use various Cadl compiler APIs to reflect on the cadl compilation and produce generated artifacts. The cadl standard library includes an emitter for OpenAPI version 3.0, but odds are good you will want to emit Cadl to another output format. In fact, one of Cadl's main benefits is how easy it is to use Cadl as a source of truth for all data shapes, and the ease of writing an emitter is a big part of that story.
+
+## Getting started
+
+Cadl emitters are a special kind of Cadl library and so have the same getting started instructions. Follow [these steps](#todo) to initialize a cadl library.
+
+## $onEmit
+
+A Cadl emitter exports a function named `$onEmit`. It receives two arguments:
```

Comment:
Not sure if this belong here or on the library setup as this is specific to emitter setup but the `$onEmit` must be exported via the `main` entry of the library.(It cannot have `import "./emitter.js", that won't work)

Reply:
I will clarify this point (I think it belongs here).